nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AnnotationServiceImpl.java

```diff
@@ -113,36 +140,70 @@ public Annotation getAnnotation(CoreSession session, String annotationId) throws
 
     @Override
     public void updateAnnotation(CoreSession session, String annotationId, Annotation annotation)
-            throws CommentNotFoundException {
+            throws CommentNotFoundException, CommentSecurityException  {
         IdRef annotationRef = new IdRef(annotationId);
         if (!session.exists(annotationRef)) {
             throw new CommentNotFoundException("The annotation " + annotationId + " does not exist.");
         }
-        DocumentModel annotationModel = session.getDocument(annotationRef);
-        Comments.annotationToDocumentModel(annotation, annotationModel);
-        if (annotation instanceof ExternalEntity) {
-            Comments.externalEntityToDocumentModel((ExternalEntity) annotation, annotationModel);
+        NuxeoPrincipal principal = session.getPrincipal();
+        if (!principal.isAdministrator() && !annotation.getAuthor().equals(principal.getName())) {
+            throw new CommentSecurityException("The user " + session.getPrincipal().getName()
+                    + " can not edit annotations of document " + annotation.getParentId());
         }
-        session.saveDocument(annotationModel);
+        CoreInstance.doPrivileged(session, s -> {
+            DocumentModel annotationModel = s.getDocument(annotationRef);
+            Comments.annotationToDocumentModel(annotation, annotationModel);
+            if (annotation instanceof ExternalEntity) {
+                Comments.externalEntityToDocumentModel((ExternalEntity) annotation, annotationModel);
+            }
+            s.saveDocument(annotationModel);
+        });
     }
 
     @Override
     public void deleteAnnotation(CoreSession session, String annotationId) throws CommentNotFoundException {
-        Framework.getService(CommentManager.class).deleteComment(session, annotationId);
+        NuxeoPrincipal principal = session.getPrincipal();
+        CoreInstance.doPrivileged(session, s -> {
+            try {
+                DocumentModel annotation = s.getDocument(new IdRef(annotationId));
+                String parentId = (String) annotation.getPropertyValue(COMMENT_PARENT_ID);
+                DocumentRef parentRef = new IdRef(parentId);
+                if (!principal.isAdministrator()
+                        && !annotation.getPropertyValue(COMMENT_AUTHOR).equals(principal.getName())
+                        && !s.hasPermission(principal, parentRef, SecurityConstants.EVERYTHING)) {
```

Comment:
Here we talk about permissions on the related document. A user having "Everything" permission on the related document is like a local administrator. Typically , community managers in collaborative use case, knowledge managers in KM use cases, departement managers in case management ones would have such a manage everything permission, so they are good candidate to do the moderation.
Note that this is a first step. In the future it would make sense to have smtg like a write comment permission and a moderate comments permission.